app/scripts/proactive/view/TaskView.js

```diff
@@ -596,7 +619,205 @@ define(
+                       hrefEyeIcon = "<i title='You cannot open non-workflows objects in the Studio' class='visu-icon disabled glyphicon glyphicon-eye-close'></i>"
+                   }
+               } else {
+                   hrefEyeIcon = "<i title='You cannot open empty objects in the Studio, select a catalog object first' class='visu-icon disabled glyphicon glyphicon-eye-close'></i>"
+                   currentVariable.Value = "";
+                   objectKind = "";
+               }
+               var markup = "<tr><td>" + currentVariable.Name + "</td><td>" + currentVariable.Value + "</td><td><button title='Select one catalog object' class='called-catalogobject-button btn' type='button' style='background-color:Transparent'><img src='images/catalog-portal.png' height='20'></button></td><td>" + objectKind + "</td><td style='text-align: center'>" + hrefEyeIcon + "</td></tr>";
+               $("#called-description-container-table").append(markup);
+           }
+           //event to activate the visualization on another tab
+           visuIconEvent();
+
+           //action on the change button to modify the catalog object in the called workflow modal
+           $(".called-catalogobject-button").click(function(event) {
+               event.preventDefault();
+               var studioApp = require('StudioApp');
+               var parentVariableValue = $(event.currentTarget).parent().parent().children()[1].innerHTML;
+               var parentVariableName = $(event.currentTarget).parent().parent().children()[0].innerHTML;
+               var parentObjectDetails = getCalledObjectDetails(parentVariableValue);
+
+               studioApp.views.catalogGetView.setKind("all", "Object");
+               studioApp.views.catalogGetView.setVarKey(event.currentTarget.getAttribute('value'));
+               studioApp.views.catalogGetView.render(parentObjectDetails["bucketName"], parentObjectDetails["objectName"]);
+               var previousZIndex = $("#catalog-get-modal").css("z-index");
+               studioApp.views.catalogGetView.setPreviousZIndex(previousZIndex);
+               var zIndexModal = parseInt($("#catalog-get-modal").parents().find(".modal").css("z-index")); // #execute-workflow-modal
+               $("#catalog-get-modal").css("z-index", (zIndexModal + 1).toString());
+               $("#catalog-get-select-button").hide();
+               $("#catalog-get-browse-button").show();
+               $('#catalog-get-modal').modal();
+
+               //action on click on the select button of the get catalog view
+               $("#catalog-get-browse-button").click(function(e) {
+                   var splitRawUrl = ($(($("#catalog-get-revisions-table .catalog-selected-row"))[0])).data("rawurl").split('/');
+                   var newBucketName = splitRawUrl[1];
+                   var newObjectName = splitRawUrl[3];
+                   var newRevisionId = "";
+                   if (splitRawUrl.length > 4) {
```

Comment:
a bit strange to access the 6th element of the array after checking that the length is greater or equal to 5

Reply:
it is correct. splitRawUrl has the following content for example.
without a revision: (length 4)
buckets,basic-examples,resources,Native_Task_Linux

with a revision: (length 6)
buckets,basic-examples,resources,Native_Task_Linux,revisions,1670256220311

Reply:
all right, but please add a comment